src/sensors/d200.hpp

```diff
@@ -0,0 +1,131 @@
+  0xb7, 0x5d, 0x10, 0xc7, 0x8a, 0x24, 0x69, 0xbe, 0xf3, 0xaf, 0xe2, 0x35,
+  0x78, 0xd6, 0x9b, 0x4c, 0x01, 0xf4, 0xb9, 0x6e, 0x23, 0x8d, 0xc0, 0x17,
+  0x5a, 0x06, 0x4b, 0x9c, 0xd1, 0x7f, 0x32, 0xe5, 0xa8
+};
+
+/// @brief serial port used by LiDAR
+#define LIDAR_SERIAL Serial1
+
+/// @brief baud rate of LiDAR module
+#define LIDAR_BAUD 230400
+
+/// @brief size of a LiDAR packet (bytes)
+const int PACKET_SIZE = 47;
+/// @brief number of points encoded in a single LiDAR packet
+const int POINTS_PER_PACKET = 12;
+/// @brief start character for a LiDAR packet
+const int START_CHAR = 0x54;
+/// @brief frame character for a LiDAR packet
+const int FRAME_CHAR = 0x2c;
+/// @brief length of LiDAR command (bytes)
+const int CMD_LEN = 8;
+/// @brief hard-coded start command for LiDAR
+const uint8_t START_CMD[] = { 0x54, 0xa0, 0x04, 0, 0, 0, 0, 0x5e };
+/// @brief hard-coded stop command for LiDAR
+const uint8_t STOP_CMD[] = { 0x54, 0xa1, 0x04, 0, 0, 0, 0, 0x4a };
+/// @brief default scanning speed (deg/s)
+const uint16_t DEFAULT_SPEED = 6 * 360;
+/// @brief minimum specified scanning speed (deg/s)
+const uint16_t MIN_SPEED = 2 * 360 + 1;
+/// @brief maximum specified scanning speed (deg/s)
+const uint16_t MAX_SPEED = 8 * 360 - 1;
+
+/// @brief struct for individual LiDAR packet
+typedef struct {
+  /// @brief current speed of LiDAR (deg/s)
+  uint16_t lidar_speed;
+  /// @brief start angle for measurements (0.01 deg)
+  uint16_t start_angle;
+  /// @brief distance and intensity measurements
+  struct {
+    /// @brief distance for point (mm)
+    uint16_t distance;
+    /// @brief signal strength for point (normalized)
+    uint8_t intensity;
+  } points[POINTS_PER_PACKET];
+  /// @brief end angle for measurements (0.01 deg)
+  uint16_t end_angle;
+  /// @brief timestamp of start of measurements (ms, wraps after 30000)
+  uint16_t timestamp;
+} LidarDataPacket;
+
+/// @brief state for LiDAR driver
+typedef struct {
+  /// @brief most recently read packet
+  LidarDataPacket latest_packet;
+  /// @brief buffer to assemble next packet
+  uint8_t buffer[PACKET_SIZE];
+  /// @brief current write index for buffer
+  uint8_t buffer_idx;
+} LidarState;
+
+/// @brief class for LiDAR driver
+class D200LD14P {
+  private:
+    LidarState state;
+
+    /// @brief compute CRC8 checksum for buffer
+    /// @param buf pointer to buffer
+    /// @param len length of buffer
+    /// @return CRC8 checksum for buffer
+    uint8_t calc_checksum(uint8_t *buf, int len);
+
+  public:
+    /// @brief constructor and initialization
+    D200LD14P();
+
+    /// @brief set the speed of the LiDAR
+    /// @param speed desired speed of LiDAR (deg/s)
+    void set_speed(uint16_t speed);
+
+    /// @brief start the LiDAR
+    void start();
+
+    /// @brief stop the LiDAR
+    void stop();
+
+    /// @brief read latest packet from LiDAR module
+    void read();
+
+    /// @brief get the latest packet
+    /// @return the latest packet
+    LidarDataPacket get_latest_packet();
```

Comment:
Make sure the end result is converted into radians, meters, and seconds, since that is what the rest of the project expects from every sensor.

You could have LidarDataPacket be the raw data as you currently have parsed, and then return a different struct with the filtered data into converted units. Could also be a good post-process step to throw out sensor time and give the teensy time instead (in seconds). 

**Addendum:** Make sure the timestamp is set when the complete packet is read, NOT when gotten in the getter here.